### PR TITLE
Add presubmit Prow jobs for kubetest2 gke deployer

### PR DESCRIPTION
**Pull request by chizhg**
Fixes https://github.com/kubernetes-sigs/kubetest2/issues/152

### CODE REVIEW

File: config/jobs/kubernetes-sigs/kubetest2/kubetest2-gke-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20210715-521d667-master
+        command:
+        - "runner.sh"
+        args:
+        - "./kubetest2-gke/ci-tests/buildupdown.sh"
+        - "--cluster-topology"
+        - "mc"
+        - "--build-strategy"
+        - "gke_make"
```
Comment (amwat) [inline config/jobs/kubernetes-sigs/kubetest2/kubetest2-gke-presubmits.yaml:40]: let's drop --build in upstream jobs since we won't have the sources.

Comment (chizhg) [reply config/jobs/kubernetes-sigs/kubetest2/kubetest2-gke-presubmits.yaml]: Got it, changed to `bazel`, if it is supported for OSS.

Comment (amwat) [reply config/jobs/kubernetes-sigs/kubetest2/kubetest2-gke-presubmits.yaml]: bazel is definitely not supported. I think it's fine to just drop build altogether and just have up down

Comment (chizhg) [reply config/jobs/kubernetes-sigs/kubetest2/kubetest2-gke-presubmits.yaml]: Thanks for the clarification. Removed.

**Review by amwat — COMMENTED** [review]
/lgtm
/approve

/hold
do we want to start with a non blocking job and then make it blocking?

**Comment by chizhg** [comment]
/unhold
I'll monitor the job runs and fix them as soon as I find any issues.

**Pull request merged**